### DISCUSSION TITLE
umi4的路由公共布局如何做？

### DISCUSSION THREAD

**Discussion by yumo-mt** [umijs/umi#11706] [Q&A] [answered]
我的配置是：
umirc.ts
`
export default defineConfig({
  antd: {},
  access: {},
  model: {},
  initialState: {},
  request: {},
  layout: {},
  routes: [{
      path: '/', 
      component: '@/layouts/index',
      routes: [
        { path: 'fuck', component: '@/pages/fuck' },
        { path: 'users', component: '@/pages/user' },
      ],
  }],
  npmClient: 'pnpm',
});

`

layout的代码
`
import { Outlet } from '@umijs/max';
export default function Page() {
  return (
    <div style={{ padding: 20 }}>
      <div>公共的头部</div>
      <Outlet />
    </div>
  );
}

`

fuck 和 user的代码都是
`
export default function fuck() {

  return (
    <div style={{ padding: 20 }}>
      <h1>index</h1>
    </div>
  );
}

`
但是页面中出现了两个公共头部。
<img width="618" alt="image" src="https://github.com/umijs/umi/assets/21261210/314ae0d1-4ce1-469d-8a27-150a68f53c87">

请教下是因为什么原因导致的。

**Comment by fz6m** [comment, marked as answer]
你是这个路由就叫 `公共的头部` 吗，使用了 `layout: {}` 布局之后会自带一个全局 layout 在展示内容的区块上，当你又自己添加了 `src/layouts/index` 时，按照布局约定，就会额外再套一层你自己定义的 `src/layout/index` 布局，也就是有两层 layout 了，具体自己排查下吧。

如还有问题，需要给一个最小复现。
